pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpruner/PartitionSegmentPruner.java

```diff
@@ -126,7 +126,8 @@ private PartitionInfo extractPartitionInfoFromSegmentZKMetadataZNRecord(String s
     }
 
     return new PartitionInfo(PartitionFunctionFactory.getPartitionFunction(columnPartitionMetadata.getFunctionName(),
-        columnPartitionMetadata.getNumPartitions()), columnPartitionMetadata.getPartitions());
+        columnPartitionMetadata.getNumPartitions(), columnPartitionMetadata.getFunctionConfig()),
+            columnPartitionMetadata.getPartitions());
```

Comment:
(code format) Can you please setup the [Pinot Style](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#intellij) and reformat the changes?

Reply:
done.